crates/rattler_package_streaming/src/reqwest/tokio.rs

```diff
@@ -131,12 +139,44 @@ pub async fn extract_conda(
     reporter: Option<Arc<dyn DownloadReporter>>,
 ) -> Result<ExtractResult, ExtractError> {
     // The `response` is used to stream in the package data
-    let reader = get_reader(url.clone(), client, expected_sha256, reporter.clone()).await?;
-    let result = crate::tokio::async_read::extract_conda(reader, destination).await?;
-    if let Some(reporter) = &reporter {
-        reporter.on_download_complete();
+    let reader = get_reader(
+        url.clone(),
+        client.clone(),
+        expected_sha256,
+        reporter.clone(),
+    )
+    .await?;
+    match crate::tokio::async_read::extract_conda(
+        reader,
+        destination,
+        crate::read::extract_conda_via_streaming,
+    )
+    .await
+    {
+        Ok(result) => {
+            if let Some(reporter) = &reporter {
+                reporter.on_download_complete();
+            }
+            Ok(result)
+        }
+        // https://github.com/conda-incubator/rattler/issues/794
+        Err(ExtractError::ZipError(zip_error))
```

Comment:
Maybe get to the inner error faster:

```
        Err(ExtractError::ZipError(ZipError::UnsupportedArchive(zip_error)))
```

Reply:
thank you! Still have to match the string because the zip crate uses `UnsupportedArchive` for multiple different error messsages